pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -109,24 +113,23 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 }
 
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
-	builder := controllerruntime.NewControllerManagedBy(m)
-	for _, nodeClass := range c.cloudProvider.GetSupportedNodeClasses() {
-		builder = builder.Watches(
-			nodeClass,
-			nodeclaimutil.NodeClassEventHandler(c.kubeClient),
-		)
-	}
-	return builder.
+	b := controllerruntime.NewControllerManagedBy(m).
 		Named("nodeclaim.disruption").
-		For(&v1.NodeClaim{}).
+		For(&v1.NodeClaim{}, builder.WithPredicates(nodeclaimutil.IsManagedPredicates(c.cloudProvider))).
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}).
+		// Note: We don't use the ManagedFilter (NodeClaim) for NodePool updates because drift should be captured when
```

Comment:
Do we even want to consider this case? Do we simplify the code if we just reason about managed resources